### PR TITLE
Added module idrac_redfish_subscription

### PR DESCRIPTION
**Pull request by TrevorSquillario**
This module allows you to create or delete Redfish Subscriptions for Metric Reports or Alerts.

Unit test included, code coverage=92%

Signed-off-by: Trevor Squillario <Trevor_Squillario@Dell.com>

### CODE REVIEW

**Comment by rajeevarakkal** [comment]
Hello,

please check the issue reported by CI/CD above.. for example

```
ERROR: Found 1 validate-modules issue(s) which need to be resolved:
ERROR: plugins/modules/idrac_redfish_subscription.py:0:0: module-invalid-version-added: DOCUMENTATION: version_added ('2.9') is not a valid collection version (see specification at https://semver.org/): invalid semantic version '2.9'. Got {'module': 'idrac_redfish_subscription', 'short_description': 'Manage Redfish Subscriptions', 'version_added': '2.9', 'description': ['This module allows you to create or delete Redfish Subscriptions for Metric Reports or Alerts.', 'HTTPS is required for the destination listening server. An example of a listening server is the Logstash HTTP Input Plugin.', 'The maximum number of subscriptions a user can create is 2.'], 'options': {'baseuri': {'description': 'IP Address of the target out-of-ba...
```

**Review by rajeevarakkal — CHANGES_REQUESTED** [review]
@TrevorSquillario - please rename this module to 'redfish_event_subscription'

**Comment by TrevorSquillario** [comment]
I'll work on these and resubmit.

**Comment by TrevorSquillario** [comment]
All test passed except the `devel` ones. I don't think this is related to my code....

They errored out with `ModuleNotFoundError: No module named 'ansible.constants'`

**Comment by rajeevarakkal** [comment]
> All test passed except the `devel` ones. I don't think this is related to my code....
> 
> They errored out with `ModuleNotFoundError: No module named 'ansible.constants'`

Thanks.

Yup, you are right, we are still figuring out.. mostly due to some ansible devel changes happening.

Will be sharing few more comments by couple of day.

**Comment by TrevorSquillario** [comment]
Any updates on this?

**Comment by anupamaloke** [comment]
> Any updates on this?

@TrevorSquillario, this is currently being planned for Sep monthly release.

**Comment by TrevorSquillario** [comment]
Done.

File: plugins/modules/redfish_event_subscription.py [hunk 1]
```diff
@@ -28,18 +28,8 @@
...
-        required: True
-    username:
-        description: Username of the target out-of-band controller.
-        type: str
-        required: True
-    password:
-        description: Password of the target out of-band controller.
-        type: str
-        required: True
+    extends_documentation_fragment:
```
Comment (sachin-apa) [inline plugins/modules/redfish_event_subscription.py:31]: @TrevorSquillario  this should be outside the **options:**

```
extends_documentation_fragment:
  - dellemc.openmanage.redfish_auth_options
options:
    destination:
        description:
          - A HTTPS URI to the destination where the events are sent.
          - HTTPS is required.
        type: str
        required: True
```

**Comment by sachin-apa** [comment]
> Done.

@TrevorSquillario: sorry if we were not clear on the addition of extends_documentation_fragment before, please correct this part of the documentation i have added a review comment on the lines.

**Comment by TrevorSquillario** [comment]
Fixed :)

**Comment by sachin-apa** [comment]
> Fixed :)

Thanks @TrevorSquillario  for the quick resolution. All tests has passed.
This is currently planned for September month release, we will merge it accordingly.

**Review by rajeevarakkal — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**